### DISCUSSION TITLE
When executing exe file: ImportError: matplotlib is required for plotting when the default backend “matplotlib” is selected when executing exe file

### DISCUSSION THREAD

**Discussion by lailacampos** [pyinstaller/pyinstaller#6177] [Help] [answered]
I created an .exe file using pyinstaller. When I run it on Pycharm, it works perfectly. But when I try to run the executable, although the program gives me the results I want, right after it ends, an error occurs.

Here's the full error:
```
Traceback (most recent call last):
  File "DT_070621.py", line 397, in <module>
  File "DT_070621.py", line 315, in arquivos_saida
  File "pandas\plotting\_core.py", line 892, in __call__
  File "pandas\plotting\_core.py", line 1814, in _get_plot_backend
  File "pandas\plotting\_core.py", line 1754, in _load_backend
ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.
[11304] Failed to execute script 'DT_070621' due to unhandled exception!
```

But when I check matplotlib version, everything seems fine.
I've also tried creating the .exe file like this:
```
pyinstaller --hidden-import=openpyxl --hidden-import=matplotlib --onefile DT_070621.py
```
But it didn't solve my problem.

Here are the imports in my script:
```
import xml.etree.ElementTree as ET
from xml.etree import ElementTree
import pandas as pd
from shapely.geometry import Point, Polygon
import numpy as np
import matplotlib.pyplot as plt
import os
```

What is going on and how do I solve it?

**Comment by rokm** [comment, marked as answer]
Try adding `--hidden-import pandas.plotting._matplotlib`.

**Reply by Golpette** [reply]
Was struggling with this for a while - thank you

**Comment by lailacampos** [comment]
For some reason I don't understand I'm unable to mark your answer as the answer to my question right now.
But it did work!
Thank you.
